internal/pkg/server/remote_es_output_integration_test.go

```diff
@@ -0,0 +1,171 @@
+	str := agentID
+	cli := cleanhttp.DefaultClient()
+	var obj map[string]interface{}
+
+	t.Logf("Fake a checkin for agent %s", str)
+	req, err := http.NewRequestWithContext(ctx, "POST", srv.baseURL()+"/api/fleet/agents/"+str+"/checkin", strings.NewReader(checkinBody))
+	require.NoError(t, err)
+	req.Header.Set("Authorization", "ApiKey "+key)
+	req.Header.Set("User-Agent", "elastic agent "+serverVersion)
+	req.Header.Set("Content-Type", "application/json")
+	res, err := cli.Do(req)
+	require.NoError(t, err)
+
+	require.Equal(t, http.StatusOK, res.StatusCode)
+	t.Log("Checkin successful, verify body")
+	p, _ := io.ReadAll(res.Body)
+	res.Body.Close()
+	err = json.Unmarshal(p, &obj)
+	require.NoError(t, err)
+
+	actionsRaw, ok := obj["actions"]
+	require.True(t, ok, "expected actions is missing")
+	actions, ok := actionsRaw.([]interface{})
+	require.True(t, ok, "expected actions to be an array")
+	require.Greater(t, len(actions), 0, "expected at least 1 action")
+	action, ok := actions[0].(map[string]interface{})
+	require.True(t, ok, "expected action to be an object")
+	typeRaw := action["type"]
+	require.Equal(t, "POLICY_CHANGE", typeRaw)
+	dataRaw := action["data"]
+	data, ok := dataRaw.(map[string]interface{})
+	require.True(t, ok, "expected data to be map")
+	policy, ok := data["policy"].(map[string]interface{})
+	require.True(t, ok, "expected policy to be map")
+	outputs, ok := policy["outputs"].(map[string]interface{})
+	require.True(t, ok, "expected outputs to be map")
+	remoteES, ok := outputs["remoteES"].(map[string]interface{})
+	require.True(t, ok, "expected remoteES to be map")
+	oType, ok := remoteES["type"].(string)
+	require.True(t, ok, "expected type to be string")
+	require.Equal(t, "elasticsearch", oType)
+	serviceToken := remoteES["service_token"]
+	require.Equal(t, nil, serviceToken)
+	remoteAPIKey := remoteES["api_key"]
+	defaultOutput, ok := outputs["default"].(map[string]interface{})
+	require.True(t, ok, "expected default to be map")
+	defaultAPIKey := defaultOutput["api_key"]
+	require.False(t, remoteAPIKey == defaultAPIKey, "expected remote api key to be different than default")
+
+}
+
+func Test_Agent_Remote_ES_Output(t *testing.T) {
+	enrollBody := `{
+	    "type": "PERMANENT",
+	    "shared_id": "",
+	    "enrollment_id": "",
+	    "metadata": {
+		"user_provided": {},
+		"local": {},
+		"tags": []
+	    }
+	}`
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// Start test server
+	srv, err := startTestServer(t, ctx, policyData)
+	require.NoError(t, err)
+
+	t.Log("Create policy with remote ES output")
+
+	var policyRemoteID = "policyRemoteID"
+	var policyDataRemoteES = model.PolicyData{
+		Outputs: map[string]map[string]interface{}{
+			"default": {
+				"type": "elasticsearch",
+			},
+			"remoteES": {
+				"type": "remote_elasticsearch",
+				// TODO start another fleet-server
+				"hosts": []string{"localhost:9200"},
```

Comment:
To have a real remote ES test, we should start another ES and refer that in hosts, and also generate a service token.
I tried to generate a remote service token with fleet-server privileges, but I got an error that it doesn't have `manage_service_account` privilege.

Reply:
We likely need to add a new ES "cluster" as part of the https://github.com/elastic/fleet-server/blob/main/dev-tools/integration/docker-compose.yml file